File: Examples/test-suite/matlab/smart_pointer_overload_runme.m
import smart_pointer_overload.*

f = Foo();
b = Bar(f);


if (f.test(3) ~= 1)
    error
end
if (f.test(3.5) ~= 2)
    error
end
if (f.test('hello') ~= 3)
    error
end

if (b.test(3) ~= 1)
    error
end
if (b.test(3.5) ~= 2)
    error
end
if (b.test('hello') ~= 3)
    error
end

File: Examples/test-suite/javascript/smart_pointer_overload_runme.js
var smart_pointer_overload = require("smart_pointer_overload");

f = new smart_pointer_overload.Foo();
b = new smart_pointer_overload.Bar(f);


if (f.test(3) != 1) {
    throw new Error;
}
if (f.test(3.5) != 2) {
    throw new Error;
}
if (f.test("hello") != 3) {
    throw new Error;
}
if (b.test(3) != 1) {
    throw new Error;
}
if (b.test(3.5) != 2) {
    throw new Error;
}
if (b.test("hello") != 3) {
    throw new Error;
}
